<launch>
    <arg name="bag" default="/home/petrito1/workspace/data/subt/dpp/2019-11-28/dfki_robot/ugv_2019-11-28-11-20-42.bag"/>

    <param name="use_sim_time" value="true"/>
    <node name="play" pkg="rosbag" type="play" args="--clock $(arg bag)" output="log">
        <remap from="rosout" to="~/rosout"/>
        <remap from="rosout_agg" to="~/rosout_agg"/>
    </node>

    <node name="joy_throttle" pkg="topic_tools" type="throttle"
          args="messages local_joy/joy 2.0 local_joy/joy_slow"/>

    <node name="joy_repeater_1" pkg="cras_cpp_common" type="joy_repeater"
          respawn="true" respawn_delay="0.5" output="screen">
        <param name="max_age" value="2.0"/>
        <param name="source_rate" value="2.0"/>
        <param name="rate" value="10.0"/>
        <param name="republish" value="false"/>
        <remap from="in" to="local_joy/joy_slow"/>
        <remap from="out" to="local_joy/joy_fast"/>
    </node>

    <node name="joy_repeater_2" pkg="cras_cpp_common" type="joy_repeater"
          respawn="true" respawn_delay="0.5" output="screen">
        <param name="max_age" value="2.0"/>
        <param name="source_rate" value="2.0"/>
        <param name="rate" value="10.0"/>
        <param name="republish" value="true"/>
        <remap from="in" to="local_joy/joy_slow"/>
        <remap from="out" to="local_joy/joy_fast_repub"/>
    </node>
</launch>